### PR TITLE
Wrapping up second test and async HTTP helpers

### PR DESCRIPTION
**Pull request by eduser25**
- Includes some helpers to create Async HTTP requests

We might add a more generic abstraction for further tests which
encompasses the HTTP request testing from multiple sources to one
destination. 
Seems a useful enough abstraction that might very well be
needed going forward.

### CODE REVIEW

File: tests/e2e/common_traffic.go [hunk 1]
```diff
@@ -74,3 +99,86 @@ func mapCurlOuput(curlOut string) map[string]string {
 	}
 	return ret
 }
+
+// HTTPAsyncResults are a storage type for async results
+type HTTPAsyncResults struct {
+	mux        *sync.Mutex // Requied to read results by the caller, as they can updating
```
Comment (eduser25) [inline tests/e2e/common_traffic.go:105]:
```suggestion
	mux        *sync.Mutex // Requied for read-write concurrency
```

File: tests/e2e/common_traffic.go [hunk 2]
```diff
@@ -74,3 +99,86 @@ func mapCurlOuput(curlOut string) map[string]string {
...
+// Convenience function
+func (res HTTPAsyncResults) withLock(f func()) {
+	res.mux.Lock()
+	defer res.mux.Unlock()
+	f()
+}
+
+// HTTPAsync is the context fed to a HTTP async thread to run HTTP queries.
+// Stores Request definition, and pointers to results, stop and wait structures.
+// Pointers are used to signal the fact that the calling application might
```
Comment (eduser25) [inline tests/e2e/common_traffic.go:128]:
```suggestion
// Pointers are used to signal the fact that the caller owns the memory allocated for 
// these structures.
```

**Review by nojnhuh — APPROVED** [review]
LGTM. I think the next step would be to try to simplify the individual tests as possible to make sure those are easy to grok. Ideally, I think anyone should be able to briefly look at this test and figure out it's simply "spin up a bunch of clients and a replicated server with SMI policies and make sure all the clients can reach the server instances."

**Comment by eduser25** [comment]
I agree with that. Next test will be simpler, and later we can rework this test to use whatever new abstractions we add.

**Pull request merged**